projects/packages/jetpack-mu-wpcom/tests/php/features/launchpad/class-launchpad-task-definitions-test.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Test class for Launchpad_Task_Lists.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+/**
+ * Test class for Launchpad_Task_Lists.
+ *
+ * @coversDefaultClass Launchpad_Task_Lists
+ */
+class Launchpad_Task_Definitions_Test extends \WorDBless\BaseTestCase {
+	/**
+	 * Set up.
+	 */
+	public function set_up() {
+		parent::set_up();
+
+		add_filter(
+			'wpcom_launchpad_extended_task_definitions',
+			function () {
+				return array(
+					'test_task1'            => array(),
+					'test_task2'            => array(),
+					'test_task3'            => array(),
+					'test_task_with_id_map' => array(
+						'id_map' => 'test_task_id_map',
```

Comment:
This setup confuses me -- from my reading of the code, we should be requiring `'test_task_id_map'` to exist, but the unit test is passing, so I feel either the code, the test, or I am missing something!

Reply:
I think the confusion came from the key and the id map being really similar. I updated those to be really different.

The idea is to test that even if we mark a task complete using the array key (in this case `test_task_with_id_map`) it will be stored in the option using the id map (`THIS_IS_THE_MAP`).

Reply:
I'll mark this as resolved because it's being discussed [here](https://github.com/Automattic/jetpack/pull/31471#discussion_r1285761913)